### PR TITLE
Movie site by Hannah Ek, Irupe Pozo Graviz and Jonas Jakobson

### PR DESCRIPTION
**Pull request by jonsjak**

### CODE REVIEW

File: code/src/components/Spinner.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import React from 'react'
+
+const Spinning = () => (
```
Comment (SofiaGerdmar) [inline code/src/components/Spinner.js:3]: Loved your spinner!

File: code/src/components/NotFound.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import { useNavigate } from 'react-router-dom';
+import notfound from '../images/notfound.png';
```
Comment (SofiaGerdmar) [inline code/src/components/NotFound.js:3]: Loved this photo! Maybe it could be centered on the page, though? Also, it seems a bit too big on screens above 1024px.

File: code/src/components/Movies.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      .catch((error) => console.log(error))
+  }, []);
+
+  return (
+    <>
+      <div className="movie-container">
+        {!loading && movies.results.map((singleMovie) => {
+          return (
+            <Link key={singleMovie.id} to={`/details/${singleMovie.id}`} className="movie-list">
+              <img className="image" src={`https://image.tmdb.org/t/p/w342/${singleMovie.poster_path}`} alt={singleMovie.original_title} />
```
Comment (SofiaGerdmar) [inline code/src/components/Movies.js:26]: I would recommend using a larger img version here, the resolution on bigger screens isn't optimal.

File: code/src/index.css [hunk 4]
```diff
@@ -1,13 +1,359 @@
...
+  flex-direction: column;
+  align-items: flex-start;
+  background-color:rgba(19, 1, 30, 0.8);
+  color: white;
+  margin-left: 10px;
+  padding-left: 9px;
+    
+  }
+
+.backButton {
```
Comment (SofiaGerdmar) [inline code/src/index.css:93]: Nice with a custom back button.

File: netlify.toml [hunk 5]
```diff
@@ -4,3 +4,8 @@
   base = "code/"
   publish = "build/"
   command = "npm run build"
+
+[[redirects]]
```
Comment (SofiaGerdmar) [inline netlify.toml:8]: Good job figuring out how to redirect from netlify's own 404 page. We borrowed that code for our site.

File: code/src/App.js [hunk 6]
```diff
@@ -1,9 +1,19 @@
...
+        <Route path="/" element={<Movies />} />
+        <Route path="/details/:movieInfo" element={<Details />} />
+        <Route path="/404" element={<NotFound />} />
+        <Route path="*" element={<Navigate to="/404" replace />} />
+
+      </Routes>
+    </BrowserRouter>
   );
-}
+}
```
Comment (SofiaGerdmar) [inline code/src/App.js:19]: Neat and tidy code!

File: code/public/index.html [hunk 7]
```diff
@@ -4,6 +4,9 @@
   <head>
     <meta charset="utf-8" />
     <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <meta property="og:title" content="Project Movies">       <!-- OG (open graph) tags are snippets of code that provide meta data about a web page, mainly to social media platforms (ex facebook, linkedin, instagram)-->
```
Comment (SofiaGerdmar) [inline code/public/index.html:7]: Nice touch adding the OG tags.

File: code/src/App.js (hunk 6 above)
Comment (Vera-Sjunnesson) [reply code/src/App.js:19]: Great idea to keep the App.js this neat

File: code/src/components/Movies.js [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+                <p className="date">Release date {singleMovie.release_date}</p>
+              </div>
+            </Link>
+          );
+        })}
+      </div>
+      {loading && (<Spinning />)}
+    </>
+  );
+};
```
Comment (Vera-Sjunnesson) [inline code/src/components/Movies.js:38]: Good structure to map through all the movies in this component outside the App.js. Plus good use of ternary operator and loading component.

File: code/src/index.css [hunk 9]
```diff
@@ -1,13 +1,359 @@
...
+  background: transparent;
+  border: none;
+  color: white;
+  font-size: 20px;
+  margin-top: 20px;
+  display: flex;
+  align-items: center;
+  cursor: pointer;
+  font-weight: 700;
+}
```
Comment (Vera-Sjunnesson) [inline code/src/index.css:128]: The css for ```.backButton```seems to have been written twice and maybe one overwrites the other. Maybe put it all together?

File: code/src/index.css [hunk 10]
```diff
@@ -1,13 +1,359 @@
...
+to {
+    transform: rotate(360deg)
+  }
+}
+
+.loading-movies p {
+  font-size: 30px;
+  text-align: center;
+  margin: 15px;
+}
```
Comment (Vera-Sjunnesson) [inline code/src/index.css:173]: Interesting to see how you built the spinner animation!

File: code/src/index.css [hunk 11]
```diff
@@ -1,13 +1,359 @@
...
+       width: 35%;
+     }
+
+   .title {
+    font-size: 22px;
+   }
+   .movie-text p {
+    font-size: 20px;
+    width: 300px;
+   }
```
Comment (Vera-Sjunnesson) [inline code/src/index.css:210]: Maybe the indentation could be improved here to be more consistent.

File: code/src/index.css [hunk 12]
```diff
@@ -1,13 +1,359 @@
...
+  position: relative;
+}
+.movie-list:hover .text-container{
+  visibility: visible;
+}
+ 
+ /*details*/
+ .backButton{
+  top: 0%;
+}
```
Comment (Vera-Sjunnesson) [inline code/src/index.css:288]: The same goes for indentation here.

**Review by Vera-Sjunnesson — APPROVED** [review]
We loved:  
- The design which was clear, user friendly and had logical navigation. 
- The loader and the 404-page which were both very inventive

Generally the site works well and is responsive. It fulfills many stretch goals and it was inspiring to watch! 

Some minor comments: 
- When viewing the site on mobile screens, the “back to movies” button is sometimes overlapping the title of the film on the movie poster and therefore not visible.
- There are error messages in the console about two images being undefined. On the other hand, the “undefined” images show up and the site functions without any problem. We believe a ternary operator when getting the image source could perhaps be the solution (how it was done in the example project). 

To summarize, great job!

Best,
Sofia & Vera
